queue/src/fiskaltrust.Middleware.Localization.QueueIT/v2/ReceiptCommandProcessorIT.cs

```diff
@@ -0,0 +1,134 @@
+
+        public async Task<ProcessCommandResponse> PointOfSaleReceipt0x0001Async(ProcessCommandRequest request)
+        {  
+            if (request.ReceiptRequest.IsVoid() || request.ReceiptRequest.IsRefund())
+            {
+                await LoadReceiptReferencesToResponse(request.ReceiptRequest, request.QueueItem, request.ReceiptResponse);
+            }
+
+            var (queue, queueIt, receiptRequest, receiptResponse, queueItem) = request;
+
+            var result = await _itSSCD.ProcessReceiptAsync(new ProcessRequest
+            {
+                ReceiptRequest = receiptRequest,
+                ReceiptResponse = receiptResponse,
+            });
+            var documentNumber = result.ReceiptResponse.GetSignaturItem(SignatureTypesIT.RTDocumentNumber);
+            var zNumber = result.ReceiptResponse.GetSignaturItem(SignatureTypesIT.RTZNumber);
+            receiptResponse.ftReceiptIdentification += $"{zNumber.Data.PadLeft(4, '0')}-{documentNumber.Data.PadLeft(4, '0')}";
+            receiptResponse.ftSignatures = result.ReceiptResponse.ftSignatures;
+            receiptResponse.InsertSignatureItems(SignaturBuilder.CreatePOSReceiptFormatSignatures(receiptResponse));
+            var journalIT = ftJournalITFactory.CreateFrom(queueItem, queueIt, new ScuResponse()
+            {
+                ftReceiptCase = receiptRequest.ftReceiptCase,
+                ReceiptNumber = long.Parse(documentNumber.Data),
+                ZRepNumber = long.Parse(zNumber.Data)
+            });
+            await _journalITRepository.InsertAsync(journalIT).ConfigureAwait(false);
+            return new ProcessCommandResponse(receiptResponse, new List<ftActionJournal>());
+        }
+
+        public async Task<ProcessCommandResponse> PaymentTransfer0x0002Async(ProcessCommandRequest request) => await Task.FromResult(new ProcessCommandResponse(request.ReceiptResponse, new List<ftActionJournal>())).ConfigureAwait(false);
+
+        public async Task<ProcessCommandResponse> PointOfSaleReceiptWithoutObligation0x0003Async(ProcessCommandRequest request) => await Task.FromResult(new ProcessCommandResponse(request.ReceiptResponse, new List<ftActionJournal>())).ConfigureAwait(false);
+
+        public async Task<ProcessCommandResponse> ECommerce0x0004Async(ProcessCommandRequest request) => await Task.FromResult(new ProcessCommandResponse(request.ReceiptResponse, new List<ftActionJournal>())).ConfigureAwait(false);
+
+        public async Task<ProcessCommandResponse> Protocol0x0005Async(ProcessCommandRequest request) => await PointOfSaleReceipt0x0001Async(request);
+
+        private async Task LoadReceiptReferencesToResponse(ReceiptRequest request, ftQueueItem queueItem, ReceiptResponse receiptResponse)
+        {
+            var queueItems = _queueItemRepository.GetByReceiptReferenceAsync(request.cbPreviousReceiptReference, request.cbTerminalID);
+            // What should we do in this case? Cannot really proceed with the storno but we
```

Comment:
Are you okay @StefanKert? What happened while you were writing this comment? :D

Reply:
Well so this case seems to be a case that we currently have in Italy. So using a storno without a receiptreference that is being stored in the current Queue. IMO there are two main cases

- We get a cbPreviousReceiptReference but can't find it: In this case I guess we should return EEEE_EEEE because you cannot void / refund sth that we don't have
- We get no cbPreviosReceiptReference but we have Refund/Void flaggings: In this case we'll proceed and the SCU should be able to handle such cases